lib/charms/layer/docker.py

```diff
@@ -0,0 +1,76 @@
+from subprocess import check_output
+from charms.docker import DockerOpts
+from charmhelpers.core import hookenv
+from charmhelpers.core.templating import render
+
+
+docker_packages = {
+    'apt': ['docker.io'],
+    'upstream': ['docker-ce'],
+    'nvidia': [
+        'docker-ce',
+        'nvidia-docker2',
+        'nvidia-container-runtime',
+        'nvidia-container-runtime-hook'
+    ]
+}
+
+
+def arch():
+    """
+    Return the package architecture as a string.
+
+    :return: String
+    """
+    return check_output(['dpkg', '--print-architecture']) \
+        .rstrip().decode('utf-8')
+
+
+def determine_apt_source():
+    """
+    :return: String docker runtime
+    """
+    config = hookenv.config
+
+    docker_runtime = config('docker_runtime')
+
+    if config('install_from_upstream'):
+        docker_runtime = 'upstream'
+
+    if docker_runtime == 'auto':
+        out = check_output(['lspci', '-nnk']).rstrip()
+        if arch() == 'amd64' \
+                and out.decode('utf-8').lower().count('nvidia') > 0:
+            docker_runtime = 'nvidia'
+        else:
+            docker_runtime = 'apt'
+
+    hookenv.log(
+        'Setting runtime to {}'.format(docker_packages))
+    return docker_runtime
+
+
+def render_configuration_template(service=False):
```

Comment:
Do we anticipate ever rendering the configuration without rendering the service? It seems everything calls this with service=True.

Reply:
I don't think this one is necessary https://github.com/juju-solutions/layer-docker/pull/135/files/0a6b0f97fa9446d0ae21dc872c2fe3d1a783fd29#diff-b72ba34de38636dd3d255bd7c29c1e73R736

But it was the old behavior so wanted to check before I changed it.